server/Auth.js

```diff
@@ -21,12 +28,267 @@ class Auth {
     }
   }
 
+  /**
+   * Inializes all passportjs strategies and other passportjs ralated initialization.
+   */
+  async initPassportJs() {
+    // Check if we should load the local strategy (username + password login)
+    if (global.ServerSettings.authActiveAuthMethods.includes("local")) {
+      passport.use(new LocalStrategy(this.localAuthCheckUserPw.bind(this)))
+    }
+
+    // Check if we should load the google-oauth20 strategy
+    if (global.ServerSettings.authActiveAuthMethods.includes("google-oauth20")) {
+      passport.use(new GoogleStrategy({
+        clientID: global.ServerSettings.authGoogleOauth20ClientID,
+        clientSecret: global.ServerSettings.authGoogleOauth20ClientSecret,
+        callbackURL: global.ServerSettings.authGoogleOauth20CallbackURL
+      }, (async function (accessToken, refreshToken, profile, done) {
+        // TODO: do we want to create the users which does not exist?
```

Comment:
This should be definitely supported as I think almost all want to have that.
Also we can read the profile variable for the groups the idp provider sends and also automatically set admin status on register (and on login, potentially adding or revoking it). For full flexibility the groupname to be expected from the IdP should be configurable.